packages/legend-application-data-cube/src/stores/LegendDataCubeDataCubeEngine.ts

```diff
@@ -215,6 +196,50 @@ export class LegendDataCubeDataCubeEngine extends DataCubeEngine {
             `Can't get query result columns. Make sure the saved query return a relation (i.e. typed TDS). Error: ${error.message}`,
           );
         }
+        const letsFuncs: V1_ValueSpecification[] = [];
+        const parameterValues = (
+          await Promise.all(
+            source.lambda.parameters.map(async (parameter) => {
+              if (
+                parameter.genericType?.rawType instanceof V1_PackageableType
+              ) {
+                const defauleValue = queryInfo.defaultParameterValues?.find(
+                  (val) => val.name === parameter.name,
+                )?.content;
+                if (defauleValue?.includes(PURE_DATE_FUNCTIONS_NAMESPACE)) {
+                  const letFunc = _deserializeLambda(
+                    await this._engineServerClient.grammarToJSON_lambda(
+                      `${LET_TOKEN} ${parameter.name} ${DataCubeQueryFilterOperator.EQUAL} ${defauleValue}`,
+                      '',
+                      undefined,
+                      undefined,
+                      false,
+                    ),
+                  );
+                  letsFuncs.push(...letFunc.body);
+                } else {
+                  const paramValue = new V1_ParameterValue();
+                  paramValue.name = parameter.name;
+                  const type = parameter.genericType.rawType.fullPath;
+                  paramValue.value =
+                    defauleValue !== undefined
+                      ? await this.parseValueSpecification(defauleValue)
+                      : {
+                          _type: V1_deserializeRawValueSpecificationType(type),
+                          value: _defaultPrimitiveTypeValue(type),
+                        };
+                  return paramValue;
+                }
+              }
+              return undefined;
+            }),
+          )
+        ).filter(isNonNullable);
+        source.lambda.body.push(...letsFuncs);
```

Comment:
why are we adding the let to the end of the expressions. 
Won't this break compilation for lambdas using the parameters within the query ?